lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -16,6 +16,35 @@ define('DOSOMETHING_GLOBAL_DEFAULT_LANG_CODE', 'en-global');
 function dosomething_global_init() {
   global $user;
 
+  // Redirect people hitting the homepage to there local version
+  if (drupal_is_front_page() && is_numeric(arg(1))) {
+
+    // Cannot use arg() as it doesn't return path prefixes
+    $path = request_path();
+    $args = explode('/', $path);
+
+    // Only apply logic to Global homepage.
+    if (!empty($path) && $args[0] != 'node') {
+      return;
+    }
```

Comment:
this seems like some wild code to determine the home page, whyyyy? what's going on?